api/src/main/java/org/pmiops/workbench/api/ProfileController.java

```diff
@@ -275,10 +275,12 @@ private User initializeUserIfNeeded() {
     if (user.getFirstSignInTime() == null) {
       // TODO(calbach): After the next DB wipe, switch this null check to
       // instead use the freeTierBillingProjectStatus.
-      if (user.getFreeTierBillingProjectName() == null) {
-        String billingProjectName = createFirecloudUserAndBillingProject(user);
-        user.setFreeTierBillingProjectName(billingProjectName);
-        user.setFreeTierBillingProjectStatusEnum(BillingProjectStatus.PENDING);
+      if (!workbenchConfigProvider.get().featureFlags.useBillingProjectBuffer && user.getFreeTierBillingProjectName() == null) {
+        if (user.getFreeTierBillingProjectName() == null) {
+          String billingProjectName = createFirecloudUserAndBillingProject(user);
```

Comment:
We still need the first part of this method call in the new codepath - which registers the user with Firecloud.